pkg/server/endpoints/node/handler.go

```diff
@@ -53,17 +55,31 @@ type Handler struct {
 	c       HandlerConfig
 	limiter Limiter
 
-	dsCache *datastoreCache
+	dsCache                          *datastoreCache
+	fetchRegistrationEntriesCache    *regentryutil.FetchRegistrationEntriesCache
+	useFetchRegistrationEntriesCache bool
 }
 
 func NewHandler(config HandlerConfig) *Handler {
 	if config.Clock == nil {
 		config.Clock = clock.New()
 	}
+	fetchX509SVIDCache, err := regentryutil.NewFetchX509SVIDCache(fetchSVIDCacheSize)
+	if err != nil {
+		config.Log.Errorf("could not create lru cache. Not using cache for FetchX509SVID", err)
```

Comment:
let's just fail hard and bail out if we can't create the cache, the likelihood of failure is small enough, and running without a cache complicates the application logic

Reply:
updated to throw the error.